arrays/arrays.c

```diff
@@ -111,13 +116,20 @@ void arr_append(Array *arr, char *element) {
 
   // Resize the array if the number of elements is over capacity
   // or throw an error if resize isn't implemented yet.
-
+  int el_size = sizeof(element) / sizeof(char);  
+  if (el_size > arr->capacity)
+  {
+    printf("Over maximum capacity.");
+    exit(1);
+  }
   // Copy the element and add it to the end of the array
-
+  arr->elements[arr->count] = element;
```

Comment:
What you're doing here is putting a pointer to the element in arr->elements[arr->count]. This might seem ok, but remember that the pointer to the input element is discarded after the function executed. It's stored in the stack which stores things that are temporarily needed to store (e.g. the input arguments to the function). What you want to do is create a new pointer and have it point to a malloc'd chunk of memory in the heap which is where you store data that you want to be accessible after a function executes. After creating this pointer in the malloc'd chunk of memory, you want to copy the contents of the input element to this new pointer. The strdup() function does this for us. Brady went into this a little bit during lecture. Note that this is also applies to the copying of the input element in the arr_append function.